packages/nimble-components/src/table/models/row-hierarchy-options-manager.ts

```diff
@@ -0,0 +1,42 @@
+import {
+    TableRecordDelayedHierarchyState,
+    type TableRecordHierarchyOptions
+} from '../types';
+
+/**
+ * Manages the hierarchy options that have been configured on a row.
+ */
+export class RowHierarchyOptionsManager {
```

Comment:
I don't think there is much value in creating this as a new class since right now everything it manages has some overlap with the existing ExpansionManager. I've removed the new class and moved the necessary logic into the ExpansionManager.